### PR TITLE
[BM/CPT] Ability to test specific jetlag PRs

### PR DESCRIPTION
**Pull request by josecastillolema**
This PR:
 - provides the ability to test specific jetlag PRs
 - incorporates variable changes from https://github.com/redhat-performance/jetlag/pull/508
 - fixes issue with step not failing according to deploy
 - introduces a new test cadence
   - have specific deploy jobs (deploy-6nodes) and remove the deploy step for the rest of them
   - introduce a weekly test (weekly-6nodes) that will deploy the cluster on Sundays and run all the tests that we have today on 4.16/ga (rest of the jobs only run by demand)
   - introduce a weekly test (weekly-6nodes) that will deploy the cluster on Wednesdays and run all the tests that we have today on 4.17/dev (rest of the jobs only run by demand)

| job                               | ocp  | build | cron       | deploys |
|-----------------------------------|------|-------|------------|---------|
| deploy-6nodes                     | 4.15 | ga    |            | x       |
| node-density-heavy-6nodes         | 4.15 | ga    |            |         |
| data-path-6nodes                  | 4.15 | ga    |            |         |
| web-burner-node-density-6nodes    | 4.15 | ga    |            |         |
| web-burner-cluster-density-6nodes | 4.15 | ga    |            |         |
| deploy-6nodes                     | 4.16 | ga    |            | x       |
| node-density-heavy-6nodes         | 4.16 | ga    |            |         |
| data-path-6nodes                  | 4.16 | ga    |            |         |
| web-burner-node-density-6nodes    | 4.16 | ga    |            |         |
| web-burner-cluster-density-6nodes | 4.16 | ga    |            |         |
| weekly-6nodes                     | 4.16 | ga    | Sundays    | x       |
| deploy-6nodes                     | 4.17 | dev   |            | x       |
| node-density-heavy-6nodes         | 4.17 | dev   |            |         |
| data-path-6nodes                  | 4.17 | dev   |            |         |
| web-burner-node-density-6nodes    | 4.17 | dev   |            |         |
| web-burner-cluster-density-6nodes | 4.17 | dev   |            |         |
| weekly-6nodes                     | 4.17 | dev   | Wednesdays | x       |

### CODE REVIEW

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
cc @jtaleric @smalleni @akrzos

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.15-nightly-x86.yaml [hunk 1]
```diff
@@ -29,6 +29,7 @@ tests:
     cluster_profile: metal-perscale-cpt
     env:
       BAREMETAL: "true"
+      CRUCIBLE: "false"
```
Comment (jtaleric) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.15-nightly-x86.yaml:32]: can we try false without ""?

Comment (josecastillolema) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.15-nightly-x86.yaml]: Will try to remove the `""` on the **_openshift-qe-installer-bm-deploy-commands.sh_** file instead

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.16-nightly-x86.yaml [hunk 2]
```diff
@@ -31,8 +31,7 @@ tests:
       BAREMETAL: "true"
       KUBE_BURNER_VERSION: 1.2.9
       NUM_WORKER_NODES: "6"
-      OCP_RELEASE_IMAGE: quay.io/openshift-release-dev/ocp-release:4.16.0-rc.9-x86_64
-      OCP_VERSION_SHORT: "4.16"
+      OCP_VERSION: latest-4.16
```
Comment (smalleni) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.16-nightly-x86.yaml:34]: This looks like a development branch job based on its name, shouldn't you set OCP_BUILD to "dev" here?

Comment (josecastillolema) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.16-nightly-x86.yaml:34]: I was under the impression that `latest-` is reserved for ga builds and `candidate-` for dev builds, is it not the case?

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.15-nightly-x86.yaml (hunk 1 above)
Comment (jtaleric) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.15-nightly-x86.yaml:32]: We should be good to go now.. I fixed it in Jetlag.

Comment (josecastillolema) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.15-nightly-x86.yaml:32]: I think removing the `""` on the **_openshift-qe-installer-bm-deploy-commands.sh_** file had already worked

Comment (josecastillolema) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.15-nightly-x86.yaml:32]: https://github.com/openshift/release/blob/eb71fba8b4eb63ecb0c46abdcc59f6dc6f606d70/ci-operator/step-registry/openshift-qe/installer/bm/deploy/openshift-qe-installer-bm-deploy-commands.sh#L45

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.16-nightly-x86.yaml (hunk 2 above)
Comment (smalleni) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.16-nightly-x86.yaml:34]: dev supports all tags here: https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp-dev-preview/ Including things like candidate-4.16, candidate-4.17 which will have the latest/last ec release and if you just use "latest" it points to the ec of the currently in dev release.

For ga you can have things like fast-4.16, stable, 4.16 or latest-4.16 or even candidate-4.16 (which I believe has the candidate z stream). Any directory on this link is fair game- https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.16-nightly-x86.yaml (hunk 2 above)
Comment (jtaleric) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.16-nightly-x86.yaml:34]: So, we should rename this from `nightly` to `candidate`?  for 4.16

Comment (jtaleric) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__metal-4.16-nightly-x86.yaml:34]: @smalleni Can you share with us how to capture a nightly build of a OCP version? For example, 4.16 nightly using the links above?

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-latest-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.16-latest-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.16-latest-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.16-latest-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.16-latest-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.16-latest-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.16-latest-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
@jtaleric ptal

**Comment by jtaleric** [comment]
Overall good with this! Did we want to chain all the workloads into a single job, to just improve the efficiency? Nice work @josecastillolema

**Review by jtaleric — APPROVED** [review]
/lgtm

only thing I want us to watch out for us the 7hr timeout.

**Pull request merged**